### PR TITLE
feature/PATRO23-41 - add remove budget test

### PR DESCRIPTION
**Pull request by DawidBski**

### CODE REVIEW

**Review by IzabelaPalubicka — APPROVED** [review]

**Review by pzajaczkowski — APPROVED** [review]

**Review by LukaszWinowicz — APPROVED** [review]

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	{
+		this.budgetRepositoryMock = new Mock<IRepository<BudgetAggregate, BudgetId>>();
+		this.instance = new HandleRemoveBudget(this.budgetRepositoryMock.Object);
+	}
+
+	/// <summary>
+	/// This unit test verifies whether calling Remove Budget Handle method on existing budget sets the budget status to Deleted.
+	/// </summary>
+	[Fact]
+	public async Task Handle_WhenCalledOnExistingBudget_ShouldSetBudgetStatusToDeleted()
```
[the remark this answers is not in the document]
Comment (mateuszgiedzinski) [reply tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs:38]: To clarify additional tests needed.

[the remark this answers is not in the document]
Comment (DawidBski) [reply tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs:38]: Ok I'll add it asap.

**Review by PiotrekkP — APPROVED** [review]

File: tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DawidBski) [reply tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs:36]: I think it's done.

File: tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		this.budgetRepositoryMock.Verify(x =>
+			x.Persist(It.Is<BudgetAggregate>(e =>
+				e.Status == Status.Deleted)));
+	}
+
+	/// <summary>
+	/// This unit test verifies whether calling Remove Budget Handle method on deleted budget throws BusinessRuleValidationException.
+	/// </summary>
+	[Fact]
+	public async Task Handle_WhenCalledOnDeletedBudget_ShouldBusinessRuleValidationException()
```
Comment (mateuszmuszynski) [inline tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs:67]: Should**Throw**BusinessRuleValidationException

File: tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		string icon = new Faker().Random.Word();
+		string description = new Faker().Lorem.Paragraph();
+		var budget = BudgetAggregate.Create(id, budgetName, userId, limit, period, icon, description);
+		this.budgetRepositoryMock.Setup(x => x.GetById(It.IsAny<BudgetId>())).ReturnsAsync(budget);
+
+		var removeBudget = new RemoveBudget(id.Value);
+		var cancellationToken = CancellationToken.None;
+
+		// Act
+		await this.instance.Handle(removeBudget, cancellationToken);
```
Comment (mateuszmuszynski) [inline tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs:55]: It's a good practice to use "sut" (system under test) variable name for class that is being tested.

Comment (DawidBski) [reply tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs:55]: I know.
image
Maybe we should consider establishing a convention in our project regarding the naming of "systems under test"?

**Review by KlaudiaGajocha — APPROVED** [review]

File: tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs (hunk 3 above)
Comment (mateuszgiedzinski) [reply tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs:55]: Okay we can leave it instance but please check other tests. We must be consistent,

Comment (DawidBski) [reply tests/unit/Intive.Patronage2023.Budget.Application.Tests/RemoveBudgetCommandHandlerTests.cs:55]: I'll make another task for this.

**Pull request merged**